src/prommis/nanofiltration/costing/diafiltration_cost_model.py

```diff
@@ -0,0 +1,406 @@
+#####################################################################################################
+# “PrOMMiS” was produced under the DOE Process Optimization and Modeling for Minerals Sustainability
+# (“PrOMMiS”) initiative, and is copyright (c) 2023-2024 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory, et al. All rights reserved.
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license information.
+#####################################################################################################
+"""
+Flowsheet costing block for diafiltration flowsheet model
+"""
+
+from pyomo.environ import Constraint, Expression, NonNegativeReals, Param, Var, units
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+
+from idaes.core import declare_process_block_class, register_idaes_currency_units
+from idaes.core.util.constants import Constants
+
+from prommis.nanofiltration.costing.diafiltration_cost_block import (
+    DiafiltrationCostingBlockData,
+)
+
+
+@declare_process_block_class("DiafiltrationCosting")
+class DiafiltrationCostingData(DiafiltrationCostingBlockData):
+    """
+    Costing block for the diafiltration flowsheet
+    """
+
+    def build_global_params(self):
```

Comment:
@mollydougher For compatibility with the `QGESSCosting` class in `ree_plant_capcost`, the global parameters should only contain the base currency and year. This way, a flowsheet will set the base currency and year based on whether the `DiafiltrationCosting` or `QGESSCosting` class is used and everything else is contained within the costing model itself.

The `QGESSCosting` class expects to find capital costs (bare equipment only, not including installation and not adjusted or annualized) as well as fixed costs related to initial construction and variable costs related to unit-specific resource usage. I'd suggest moving the `_build_common_global_params()` call inside your `build_process_costs` method as they are specific to your module. Similarly, `factor_total_investment`, `factor_maintenance_labor_chemical`, `factor_capital_annualization`, and `capital_recovery_factor` should move inside the `build_process_costs` method. It might make sense to move `density` and `specific_gravity` inside `cost_pump` if that is the only place they are used. If `electricity_cost` is used in only a couple places, it's probably okay to redefine it.

One last note, I don't see where `variable_operating_cost` is being calculated, although it appears in the aggregate cost section. Costs from resources (electricity, chemicals, water, waste streams with a known disposal cost, ...) should be included in variable OPEX, while costs from any materials or other items related to construction including membrane replacement can be fixed OPEX. The QGESS class already includes estimates of maintenance, labor, admin, taxes, etc., as well as installation costs for equipment.

Reply:
Thanks for this feedback, @bpaul4. Everything makes sense. After moving around where those variables are defined, I ran into some DOF issues that I'm still debugging. I should have a commit up sometime tomorrow to address these and your remaining comments.